harmony-api/src/services/postService.ts

```diff
@@ -56,12 +56,63 @@ export const getTrendingPosts = async (
   return postRepo.getTrendingPosts(requester)
 }
 
-export const getFollowingPosts = async (userId?: string) => {
+export const getFollowingUserPosts = async (userId?: string) => {
   if (!userId) {
     return []
   }
 
-  return await postRepo.getFollowingPosts(userId)
+  return await postRepo.getFollowingUserPosts(userId)
+}
+
+export const getFollowingArtistPosts = async (userId?: string) => {
+  if (!userId) {
+    return []
+  }
+
+  return await postRepo.getFollowingArtistPosts(userId)
+}
+
+export const getFollowingSongPosts = async (userId?: string) => {
+  if (!userId) {
+    return []
+  }
+
+  return await postRepo.getFollowingSongPosts(userId)
+}
+
+export const getFollowingAlbumPosts = async (userId?: string) => {
+  if (!userId) {
+    return []
+  }
+
+  return await postRepo.getFollowingAlbumPosts(userId)
+}
+
+export const getAllFollowingPosts = async (userId?: string) => {
+  if (!userId) {
+    return []
+  }
+
+  const userPosts = await getFollowingUserPosts(userId)
```

Comment:
In case you were curious, you can await many promises with await Promise.all(Promise[]), dont need any changes here though